desktop/logger/index.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Credit: Modified from `winston-namespace` by @SetaSouto:
+ * 	https://github.com/SetaSouto/winston-namespace
+ *
+ * Key changes:
+ *
+ * 1. Use a custom base configuration to avoid overriding the default configuration when
+ * initializing each logger object. This maintains the same logger initialization API
+ * as the prior logging solution (`debug` module).
+ *
+ * 2. Maintain prior log filtering behavior with the `DEBUG` environment variable
+ * (instead of the `LOG_NAMESPACES` variable used in winston-namespace).
+ */
+
+/**
+ * External dependencies
+ */
+const { app } = require('electron');
+const path = require('path');
+const { createLogger, format, transports } = require('winston');
+
+/**
+ * Internal dependencies
+ */
+const namespaces = require('./namespaces');
+
+/**
+ * Module variables
+ */
+const maxFiles = 3;
+const maxsize = 15000000;
+
+module.exports = (namespace, options) => {
+  if (!options || typeof options !== 'object') options = {};
+
+  const formatMessageWithMeta = (info) => {
+    const args = info[Symbol.for('splat')];
```

Comment:
what's the reason for a `Symbol` here and what's the reason for `splat`?

Reply:
It's been a while so my recollection is a little fuzzy - wish I added a comment when I initially wrote this. 😅  Per the Winston docs, `Symbol.for('splat')` is how splat arguments are extracted. I believe this piece of logic handles the case where splat arguments could be populated with an array like `[undefined]` or some such, leading to some funky output.

I _think_ this is moot if splatting is entirely not used (e.g. `log.info("message: %s", x)`), although you may encounter it even with `log.info("message: ", x)`, but not 100 percent sure -  I may have been working around some default behavior. You may need to verify output with/without this piece of logic.